### PR TITLE
ci: Fix path of contrib cache

### PR DESCRIPTION
**Pull request by daniel-larraz**
The cache for contrib deps was not being saved because the path was incorrect.

### CODE REVIEW

File: .github/actions/setup-cache/action.yml [hunk 1]
```diff
@@ -54,14 +54,13 @@ runs:
       id: contrib-cache
       uses: actions/cache@v4
       with:
-        path: deps/install
+        path: deps
```
Comment (mpreiner) [inline .github/actions/setup-cache/action.yml:57]: Originally, dependencies were installed to `deps/install`, is this not the case anymore with the contrib/get-* scripts? Caching the whole deps directory will also include all build artifacts, which can be quite big.

Comment (daniel-larraz) [reply .github/actions/setup-cache/action.yml:57]: > Is this not the case anymore with the contrib/get-* scripts?

That's right.

Notice, however, this `deps` directory is not the one within the `build` directory. This `deps` directory is created at the root, so only artifacts downloaded with the contrib/get-* scripts are put there.

Comment (mpreiner) [reply .github/actions/setup-cache/action.yml:57]: Yes, I'm aware of that. I don't think that we want to cache all build artifacts for these dependencies, but only the binaries/libraries we need.

Comment (daniel-larraz) [reply .github/actions/setup-cache/action.yml]: Changed. Only `deps/bin` is cached now.

**Review by mpreiner — APPROVED** [review]

**Pull request merged**